teuthology/dispatcher/supervisor.py

```diff
@@ -0,0 +1,204 @@
+        # dies (e.g. because of a restart)
+        result_proc = subprocess.Popen(args=args, preexec_fn=os.setpgrp)
+        log.info("teuthology-results PID: %s", result_proc.pid)
+        # Remove unnecessary logs for first and last jobs in run
+        for f in os.listdir(job_config['archive_path']):
+            os.remove(os.path.join(job_config['archive_path'], f))
+        os.rmdir(job_config['archive_path'])
+        return
+
+    log.info('Running job %s', job_config['job_id'])
+
+    arg = [
+        os.path.join(teuth_bin_path, 'teuthology'),
+    ]
+    # The following is for compatibility with older schedulers, from before we
+    # started merging the contents of job_config['config'] into job_config
+    # itself.
+    if 'config' in job_config:
+        inner_config = job_config.pop('config')
+        if not isinstance(inner_config, dict):
+            log.warn("run_job: job_config['config'] isn't a dict, it's a %s",
+                     str(type(inner_config)))
+        else:
+            job_config.update(inner_config)
+
+    if verbose or job_config['verbose']:
+        arg.append('-v')
+
+    arg.extend([
+        '--owner', job_config['owner'],
+        '--archive', job_config['archive_path'],
+        '--name', job_config['name'],
+    ])
+    if job_config['description'] is not None:
+        arg.extend(['--description', job_config['description']])
+    job_archive = os.path.join(job_config['archive_path'], 'orig.config.yaml')
+    arg.extend(['--', job_archive])
+
+    log.debug("Running: %s" % ' '.join(arg))
+    p = subprocess.Popen(args=arg)
+    log.info("Job archive: %s", job_config['archive_path'])
+    log.info("Job PID: %s", str(p.pid))
+
+    if teuth_config.results_server:
+        log.info("Running with watchdog")
+        try:
+            run_with_watchdog(p, job_config)
+        except Exception:
+            log.exception("run_with_watchdog had an unhandled exception")
+            raise
+    else:
+        log.info("Running without watchdog")
+        # This sleep() is to give the child time to start up and create the
+        # archive dir.
+        time.sleep(5)
+        p.wait()
+
+    if p.returncode != 0:
+        log.error('Child exited with code %d', p.returncode)
+    else:
+        log.info('Success!')
+    if 'targets' in job_config:
+        unlock_targets(job_config)
+
+
+def reimage_machines(job_config):
+    # Reimage the targets specified in job config
+    # and update their keys in config after reimaging
+    ctx = create_fake_context(job_config)
+    # change the status during the reimaging process
+    report.try_push_job_info(ctx.config, dict(status='waiting'))
+    targets = job_config['targets']
+    reimaged = reimage_many(ctx, targets, job_config['machine_type'])
+    ctx.config['targets'] = reimaged
+    # change the status to running after the reimaging process
+    report.try_push_job_info(ctx.config, dict(status='waiting'))
```

Comment:
The comment above this line states it should set status to 'running' but you're trying to set it to 'waiting'

Reply:
Oops, changed it.